lib/assertions/directory_content.js

```diff
@@ -0,0 +1,207 @@
+module.exports = function (chai, utils) {
+
+	var Assertion = chai.Assertion;
+	var flag = utils.flag;
+	var assert = chai.assert;
+
+	var readdir = require('readdir-enhanced');
+	var format = require('util').format;
+
+	//-------------------------------------------------------------------------------------------------------------
+
+	function directoryContentAssertion (label, expectedContents, msg) {
+		var dir = flag(this, 'fs.path');
+		var deep = !!flag(this, 'deep');
+		var actualContents = flag(this, 'object');
+		var contains = !!flag(this, 'contains');
+
+		var preMsg = '';
+		if (msg) {
+			flag(this, 'message', msg);
+			preMsg = msg + ': ';
+		}
+
+		new chai.Assertion(expectedContents, preMsg + 'expected-value').is.an('array');
+
+		actualContents.sort();
+		expectedContents.sort();
+
+		if (deep) {
+			label = 'deep ' + label;
+		}
+
+		var pass, failMsg, failNegateMsg;
+		try {
+			if (contains) {
+				failMsg = format("expected '%s' to contain %s #{exp}", dir, label);
+				failNegateMsg = format("expected '%s' to not contain %s #{exp}", dir, label);
+				new chai.Assertion(actualContents, msg).contains.members(expectedContents, msg);
+			} else {
+				failMsg = format("expected '%s' to have the same %s as #{exp}", dir, label);
+				failNegateMsg = format("expected '%s' to not have the same %s as #{exp}", dir, label);
+				new chai.Assertion(actualContents, msg).has.members(expectedContents, msg);
+			}
+			pass = true;
+		}
+		catch (e) {
+			pass = false;
+		}
```

Comment:
This is probably fine the way it is since I don't see anything in the `.members` logic that would throw anything other than an `AssertionError`, but it still might be a good idea for the purposes of future-proofing to only set `pass = false` on line 47 if the type of error thrown is an `AssertionError`. If the error thrown isn't an `AssertionError`, then it indicates an unexpected error instead of a failed assertion, and should be re-thrown. Example: https://github.com/chaijs/chai/blob/93905a7441242565d36e805549767c3d2bd35461/lib/chai/core/assertions.js#L308